### ISSUE TITLE
compressor fails on Promise.prototype.catch

### ISSUE THREAD

**Issue by mdvorscak** [YUICompressor-NET/YUICompressor.NET#29]
When attempting to compress:
```js
var promise = new Promise(function (resolve, reject) {
    reject(new Error('Test'));
} );
promise
    .then(function(){console.log('success')})
    .catch(function(e) {console.error(e)});
```

I get the error: `Message	"missing name after . operator"`

Full stack trace:
```
 at Yahoo.Yui.Compressor.CustomErrorReporter.Error(String message, String sourceName, Int32 line, String lineSource, Int32 lineOffset) in C:\projects\yuicompressor-net\Code\Yahoo.Yui.Compressor\CustomErrorReporter.cs:line 57
   at EcmaScript.NET.Parser.AddError(String messageId) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 96
   at EcmaScript.NET.Parser.ReportError(String messageId) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 103
   at EcmaScript.NET.Parser.memberExprTail(Boolean allowCallSyntax, Node pn) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1963
   at EcmaScript.NET.Parser.memberExpr(Boolean allowCallSyntax) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1898
   at EcmaScript.NET.Parser.unaryExpr() in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1736
   at EcmaScript.NET.Parser.mulExpr() in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1649
   at EcmaScript.NET.Parser.addExpr() in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1629
   at EcmaScript.NET.Parser.shiftExpr() in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1607
   at EcmaScript.NET.Parser.relExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1577
   at EcmaScript.NET.Parser.eqExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1525
   at EcmaScript.NET.Parser.bitAndExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1514
   at EcmaScript.NET.Parser.bitXorExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1503
   at EcmaScript.NET.Parser.bitOrExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1492
   at EcmaScript.NET.Parser.andExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1480
   at EcmaScript.NET.Parser.orExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1468
   at EcmaScript.NET.Parser.condExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1451
   at EcmaScript.NET.Parser.assignExpr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1433
   at EcmaScript.NET.Parser.expr(Boolean inForInit) in C:\projects\ecmascript-net\Code\EcmaScript.NET\Parser.cs:line 1422
```

Same issue as reported here: https://github.com/yui/yuicompressor/issues/203

I will also forward this along to the EcmaScript.NET project (since that is where the source issue is coming from)

**Comment by freeranger** [comment]
Hi - it is the ECMAScript parser that would need to change.
Have you tried the workaround mentioned in the issue you reference?
Or there is a further reference to another issue which says to use ['catch'](errorCallback) instead of .catch(errorCallback).

Neither are great, but better than nothing if no fix is forthcoming.

**Comment by mdvorscak** [comment]
Yeah, I forwarded the issue along here: https://github.com/PureKrome/EcmaScript.NET/issues/4 . I posted the issue here as more of a tracker to update the EcmaScript.NET dependency when the issue is fixed in their library. 

I can certainly use a workaround in my case. However, we provide the compressor as a service to other business entities in our company that are not as well versed in JS. Communicating this workaround would inherently cost a lot. Fixing the issue in third party libraries seems to be the quickest solution to our problem.

**Comment by freeranger** [comment]
Indeed - I have no doubt a PR would be gratefully received on the ecmascript project if you want to put one together.

**Comment by mdvorscak** [comment]
Seems one already exists https://github.com/PureKrome/EcmaScript.NET/pull/7 , I prefer to see what @PureKrome thinks about it before making my own.

**Comment by freeranger** [comment]
Fair enough, didn’t see that - hey @PureKrome pull your finger out m8 :)

**Comment by PureKrome** [comment]
Problem is: i don't want to install these older versions of .NET ... :( which means I'm hesitant to accept without manual testing, first.

That ecma was targetting .NET 2.0 :/

Yes, not a good excuse ... I totally get that. Wish I could just have some Docker image with .NET 2.0 so i can dev against that....

**Comment by PureKrome** [comment]
Maybe if we update the two libraries to .NET Standard 1.0 (which is an equiv of .NET 4.5 and higher) then I can resolve this problem...

**Comment by freeranger** [comment]
I guess you could keep around the current version as the last to support 2.0 and increase the major version.
Support for 4.5 had ended so no point supporting that. 4.6.2 will be supported intil 2020 which is when Windows 7 support goes, so I’d say go for .NET Framework 4.6.2, Standard 1.0, Core 1.0

https://support.microsoft.com/en-gb/help/13853/windows-lifecycle-fact-sheet

**Comment by PureKrome** [comment]
Closing this as it should be fixed in #33 which now references ECMAScript library 2.0.0 which should have the fix in it.
